ICSharpCode.Decompiler/CSharp/SequencePointBuilder.cs

```diff
@@ -348,26 +371,64 @@ internal static bool HasUsableILRange(ILInstruction inst)
 								newList[newList.Count - 1] = last;
 							}
 						}
-					} else if (sequencePoint.Offset > pos) {
-						// insert hidden sequence point in the gap.
-						var hidden = new DebugInfo.SequencePoint();
-						hidden.Offset = pos;
-						hidden.EndOffset = sequencePoint.Offset;
-						hidden.SetHidden();
-						newList.Add(hidden);
 					}
+
 					newList.Add(sequencePoint);
 					pos = sequencePoint.EndOffset;
 				}
+				// Add a hidden sequence point to account for the epilog of the function
 				if (pos < function.CodeSize) {
 					var hidden = new DebugInfo.SequencePoint();
 					hidden.Offset = pos;
 					hidden.EndOffset = function.CodeSize;
 					hidden.SetHidden();
 					newList.Add(hidden);
 				}
+
+				
+				List<int> sequencePointCandidates = function.MoveNextFunction == null ? function.SequencePointCandidates : function.MoveNextFunction.SequencePointCandidates;
+				if (sequencePointCandidates.Count > 2) {
```

Comment:
Why `> 2` and not `> 0` ?
Even a single candidate might help us move one of ours?

Reply:
Good catch. It's a copy and replace typo. Should have been if newList.Count > 2